sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/collectionOperations.scala

```diff
@@ -287,3 +288,173 @@ case class ArrayContains(left: Expression, right: Expression)
 
   override def prettyName: String = "array_contains"
 }
+
+/**
+ * Creates a String containing all the elements of the input array separated by the delimiter.
+ */
+@ExpressionDescription(
+  usage = """
+    _FUNC_(array, delimiter[, nullReplacement]) - Concatenates the elements of the given array
+      using the delimiter and an optional string to replace nulls. If no value is set for
+      nullReplacement, any null value is filtered.""",
+  examples = """
+    Examples:
+      > SELECT _FUNC_(array('hello', 'world'), ' ');
+       hello world
+      > SELECT _FUNC_(array('hello', null ,'world'), ' ');
+       hello world
+      > SELECT _FUNC_(array('hello', null ,'world'), ' ', ',');
+       hello , world
+  """, since = "2.4.0")
+case class ArrayJoin(
+    array: Expression,
+    delimiter: Expression,
+    nullReplacement: Option[Expression]) extends Expression with ExpectsInputTypes {
+
+  def this(array: Expression, delimiter: Expression) = this(array, delimiter, None)
+
+  def this(array: Expression, delimiter: Expression, nullReplacement: Expression) =
+    this(array, delimiter, Some(nullReplacement))
+
+  override def inputTypes: Seq[AbstractDataType] = if (nullReplacement.isDefined) {
+      Seq(ArrayType(StringType), StringType, StringType)
+    } else {
+      Seq(ArrayType(StringType), StringType)
+    }
```

Comment:
nit: indent?

Reply:
I don't think the indent is wrong since this is for the if...else and not for the method itself

Reply:
Hmm, I think the indent is 2 spaces in this case. For example, [namedExpressions.scala#L170-L174](https://github.com/mgaido91/spark/blob/e52ff856d42adc5af2e2b2593c2e63d5c3f3a205/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/namedExpressions.scala#L170-L174) or [regexpExpressions.scala#L46-L51](https://github.com/mgaido91/spark/blob/e52ff856d42adc5af2e2b2593c2e63d5c3f3a205/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/regexpExpressions.scala#L46-L51).

Reply:
ops, you are right....I am not sure where I saw it differently...maybe I just got confused...sorry, I am fixing it